### PR TITLE
[CentralDashboard] Update Dockerfile to run all tests apart from built image

### PR DESCRIPTION
**Pull request by prodonjs**
This ensures that all unit tests are run during the image build process, which is invoked during pre/post submit tests.

/area centraldashboard
/area front-end
/cc @avdaredevil

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/3541)

### CODE REVIEW

File: components/centraldashboard/Dockerfile [hunk 1]
```diff
@@ -1,18 +1,35 @@
-FROM node:10
+# Step 1: Builds and tests
+FROM node:10-alpine AS test
```
Comment (avdaredevil) [inline components/centraldashboard/Dockerfile]: What's the difference?

**Comment by avdaredevil** [comment]
/lgtm
/approve

Feel free to answer the question after the merge, or in a DM. Thanks!

File: components/centraldashboard/Dockerfile (hunk 1 above)
Comment (prodonjs) [reply components/centraldashboard/Dockerfile]: Sorry, didn't see this comment! Alpine is a smaller image that the base node image (https://hub.docker.com/_/node/) so this reduces the size of our build quite a bit.

Comment (avdaredevil) [reply components/centraldashboard/Dockerfile]: Gotcha, thanks!

**Comment by prodonjs** [comment]
@avdaredevil I made some additional changes to shrink the image size since it seemed wildly larger than some of the other UI components. We're now down from [400MB](https://pantheon.corp.google.com/gcr/images/kubeflow-images-public/GLOBAL/centraldashboard@sha256:be54a22bcd3cbc3bff8a5d2fcf9f7da70b5ab042e86d4ec4adc876fbc565aa3a/details?tab=info) to about [50MB](https://pantheon.corp.google.com/gcr/images/kubeflow-images-public/GLOBAL/centraldashboard@sha256:4e7b4e7915253e780f8f7e5f76759a1649d2c3c967bfe2fa283aee7b486fc7c9/details?tab=info).

**Comment by avdaredevil** [comment]
Woohoo! That's great news 💯 

/lgtm
/approve

**Pull request merged**